cmake/SetLibFuzzerPath.cmake

```diff
@@ -0,0 +1,69 @@
+function(SetHwArchString outvar)
+  if (CMAKE_SIZEOF_VOID_P EQUAL 4)
+    set(hw_arch "i386")
+  elseif (CMAKE_SIZEOF_VOID_P EQUAL 8)
+    set(hw_arch "x86_64")
+  else ()
+    message(FATAL_ERROR "Unsupported architecture.")
+  endif ()
+  set(${outvar} ${hw_arch} PARENT_SCOPE)
+endfunction()
+
+# The function sets the given variable in a parent scope to a
+# value with path to a path to libclang_rt.fuzzer_no_main [1]
+# library. Function raises a fatal message if C compiler is not
+# Clang.
+#
+# $ clang-15 -print-file-name=libclang_rt.fuzzer_no_main-x86_64.a
+# $ /usr/lib/llvm-15/lib/clang/15.0.7/lib/linux/libclang_rt.fuzzer_no_main-x86_64.a
+#
+# 1. https://llvm.org/docs/LibFuzzer.html#using-libfuzzer-as-a-library
+function(SetLibFuzzerPath outvar)
+  if (NOT CMAKE_C_COMPILER_ID STREQUAL "Clang")
+    message(FATAL_ERROR "C compiler is not a Clang")
+  endif ()
+
+  SetHwArchString(HW_ARCH)
+  if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    set(lib_name "libclang_rt.fuzzer_no_main-${HW_ARCH}.a")
+  else()
+    message(FATAL_ERROR "Unsupported system: ${CMAKE_SYSTEM_NAME}")
+  endif()
+
+  execute_process(COMMAND ${CMAKE_C_COMPILER} "-print-file-name=${lib_name}"
```

Comment:
It's better to check that the result name isn't the same parameter for the `-print-file-name` flag.
```sh
clang -print-file-name=noname
noname
```

This will help to detect incorrect library name.